skimage/util/noise.py

```diff
@@ -2,8 +2,93 @@
 from .dtype import img_as_float
 
 
-__all__ = ['random_noise']
+__all__ = ['random_noise', 'blue_noise']
 
+def blue_noise(shape, radius, k=30, seed=None):
+    """
+    Generate blue noise over a two-dimensional rectangle of size (width,height)
+
+    Parameters
+    ----------
+
+    shape : tuple
+        Two-dimensional domain (width x height) 
+    radius : float
+        Minimum distance between samples
+    k : int, optional
+        Limit of samples to choose before rejection (typically k = 30)
+    seed : int, optional
+        If provided, this will set the random seed before generating noise,
+        for valid pseudo-random comparisons.
+
+    References
+    ----------
+
+    .. [1] Fast Poisson Disk Sampling in Arbitrary Dimensions, Robert Bridson,
+           Siggraph, 2007. :DOI:`10.1145/1278780.1278807`
+    """
+
+    def sqdist(a, b):
+        """ Squared Euclidean distance """
+        dx, dy = a[0] - b[0], a[1] - b[1]
+        return dx * dx + dy * dy
+
+    def grid_coords(p):
+        """ Return index of cell grid corresponding to p """
+        return int(floor(p[0] / cellsize)), int(floor(p[1] / cellsize))
+
+    def fits(p, radius):
+        """ Check whether p can be added to the queue """
+
+        radius2 = radius*radius
+        gx, gy = grid_coords(p)
+        for x in range(max(gx - 2, 0), min(gx + 3, grid_width)):
+            for y in range(max(gy - 2, 0), min(gy + 3, grid_height)):
+                g = grid[x + y * grid_width]
+                if g is None:
+                    continue
+                if sqdist(p, g) <= radius2:
+                    return False
+        return True
+
+    # When given a seed, we use a private random generator in order to not
+    # disturb the default global random generator
+    if seed is not None:
+        from numpy.random.mtrand import RandomState
+        rng = RandomState(seed=seed)
+    else:
+        rng = np.random
+    
+    width, height = shape
+    cellsize = radius / sqrt(2)
+    grid_width = int(ceil(width / cellsize))
+    grid_height = int(ceil(height / cellsize))
+    grid = [None] * (grid_width * grid_height)
+
+    p = rng.uniform(0,shape, 2)
+    queue = [p]
+    grid_x, grid_y = grid_coords(p)
+    grid[grid_x + grid_y * grid_width] = p
+
+    while queue:
+        qi = rng.randint(len(queue))
+        qx, qy = queue[qi]
+        queue[qi] = queue[-1]
+        queue.pop()
+        for _ in range(k):
+
+            # WARNING: p is not uniform sampled but we can live with it here
```

Comment:
There is still this line I don't understand, and you didn't reply to me about my suggestion

Reply:
Sorry, I did not see it. I need to pick k points around p in a torus (r,2r) and I used uniform polar coordinates introducing a bias when converted into Cartesian coordinates.  I'll fix it and remove the comment.

Reply:
And the implementation by @emulbreh is right actually.

Reply:
I'm not sure about the suggestion you're referring to

Reply:
LGTM.